cmd/gateway/commands.go

```diff
@@ -168,6 +181,13 @@ func createStaticModeCommand() *cobra.Command {
 			"equal, it will choose the resource that appears first in alphabetical order by {namespace}/{name}.",
 	)
 
+	cmd.Flags().StringVar(
+		&configCRDName,
+		"nginx-gateway-config-name",
+		"nginx-gateway-config",
```

Comment:
The user can still see it in the deployment spec. My thoughts were to handle the case where it may accidentally be omitted from that spec.

Reply:
>  My thoughts were to handle the case where it may accidentally be omitted from that spec.

I think it is better to fail fast in this case

Reply:
What do you mean by fail fast? Do you think the controller should fail to start if this arg is not provided?

Reply:
> I think it is better to fail fast in this case

this was because my wrong assumption we need to require the resource is set.

I think if we don't require that resource, the default should be "" .
If the user wants to configure it, they explicitly choose a name, so that it is explicitly defined (rather than assumed by default)